### PR TITLE
[Merged by Bors] - Implement SMIP-0002: Remove vesting vault cliff

### PR DESCRIPTION
**Pull request by lrettig**
## Motivation

See https://github.com/smfoundation/smips/blob/main/proposals/smip-0002-remove-vault-vesting-clff.md

See also https://github.com/spacemeshos/economics/pull/40

## Description

Removes vesting cliff for vesting vault

## Test Plan

No new tests required. Updates existing tests.

## TODO

- [x] Explain motivation or link existing issue(s)
- [x] Test changes and document test plan
- [ ] Update documentation as needed
- [x] Update [changelog](../CHANGELOG.md) as needed

### CODE REVIEW

**Comment by pigmej** [comment]
@lrettig imo we definitely need changelog mention here too.

**Comment by lrettig** [comment]
Yup will do!

**Review by dshulyak — APPROVED** [review]

**Comment by lrettig** [comment]
@dshulyak would you mind taking one more look? Thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -16,11 +16,15 @@ See [RELEASE](./RELEASE.md) for workflow instructions.
 
 * [#5803](https://github.com/spacemeshos/go-spacemesh/pull/5803) Fixed PoST verifiers autoscaling for 1:N setups.
 
+* [#5807](https://github.com/spacemeshos/go-spacemesh/pull/5807) Implement SMIP-0002: remove vesting vault cliff.
```
Comment (fasmat) [inline CHANGELOG.md]: We just released v1.4.5, you will probably need to move the changelog 🙂

**Review by fasmat — APPROVED** [review]

**Review by dshulyak — APPROVED** [review]

**Comment by lrettig** [comment]
bors merge

**Pull request closed without merging**